src/Essentials/src/Platform/WindowStateManager.ios.cs

```diff
@@ -84,23 +81,47 @@ class WindowStateManagerImplementation : IWindowStateManager
 
 		public UIWindow? GetCurrentUIWindow()
 		{
-			// This call site is reachable on: 'iOS' 10.0 and later.0 'UIApplication.KeyWindow.get' is unsupported on: 'ios' 13.0 and later.
-			var window = UIApplication.SharedApplication.KeyWindow;
+			var window = GetKeyWindow();
 
 			if (window != null && window.WindowLevel == UIWindowLevel.Normal)
 				return window;
 
 			if (window == null)
 			{
-				// This call site is reachable on: 'iOS' 10.0 and later. 'UIApplication.Windows.get' is unsupported on: 'ios' 15.0 and later.
-				window = UIApplication.SharedApplication
-					.Windows
+				window = GetWindows()?
 					.OrderByDescending(w => w.WindowLevel)
 					.FirstOrDefault(w => w.RootViewController != null && w.WindowLevel == UIWindowLevel.Normal);
 			}
-#pragma warning restore CA1416
 
 			return window;
 		}
+
+		static UIWindow? GetKeyWindow()
+		{
+			// if we have scene support, use that
+			if (OperatingSystem.IsIOSVersionAtLeast(13) || OperatingSystem.IsMacCatalystVersionAtLeast(13))
+			{
+				var scenes = UIApplication.SharedApplication.ConnectedScenes;
+				var windowScene = scenes.ToArray<UIWindowScene>().FirstOrDefault();
+				return windowScene?.Windows.FirstOrDefault();
+			}
+
+			// use the windows property (up to 13.0)
+			return UIApplication.SharedApplication.KeyWindow;
+		}
+
+		static UIWindow[]? GetWindows()
+		{
+			// if we have scene support, use that
+			if (OperatingSystem.IsIOSVersionAtLeast(13) || OperatingSystem.IsMacCatalystVersionAtLeast(13))
```

Comment:
I dropped this to 13 as this API is supposed to be used on iOS 13+ - the 15 is the max the `UIApplication.SharedApplication.Windows` API is used. So having this as 13+ means that it will never fall back on 14 and 15 - and that is what we want as we use scenes on 13.